apiserver/facades/agent/upgrader/unitupgrader.go

```diff
@@ -125,6 +118,82 @@ func (u *UnitUpgraderAPI) DesiredVersion(ctx context.Context, args params.Entiti
 	return params.VersionResults{Results: result}, nil
 }
 
+// SetTools is responsible for updating a a set of entities reported agent
+// version.
+func (u *UnitUpgraderAPI) SetTools(
```

Comment:
`SetAgentBinaryVersion` or something, not Tools

Reply:
This is a facade. I am not refactoring facades and versions at the moment.